src/NServiceBus.Core.Tests/Transports/PushRuntimeSettingsTests.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus.Core.Tests.Transports
+{
+    using System;
+    using NUnit.Framework;
+    using Transport;
+
+    [TestFixture]
+    public class PushRuntimeSettingsTests
+    {
+        [Test]
+        public void Should_default_concurrency_to_num_processors()
+        {
+            Assert.AreEqual(Math.Max(2, Environment.ProcessorCount), new PushRuntimeSettings().MaxConcurrency);
```

Comment:
I contemplated that but called YAGNI on that for now. The test at least makes sure that we won't change the algorithm by accident?